test_compileall2.py

```diff
@@ -384,6 +384,177 @@ def test_ignore_symlink_destination(self):
         self.assertTrue(os.path.isfile(allowed_bc))
         self.assertFalse(os.path.isfile(prohibited_bc))
 
+    def test_hardlink_deduplication_bad_args(self):
+        # Bad arguments combination, hardlink deduplication make sense
+        # only for more than one optimization level
+        with self.assertRaises(ValueError):
+            compileall.compile_dir(self.directory, quiet=True, optimize=0, hardlink_dupes=True)
+
+    def test_hardlink_deduplication_same_bytecode_all_opt(self):
+        # 'a = 0' produces the same bytecode for all optimization levels
+        path = os.path.join(self.directory, "test", "same_all")
+        os.makedirs(path)
+
+        simple_script = script_helper.make_script(path, "test_same_bytecode", "a = 0")
+        pyc_opt0 = importlib.util.cache_from_source(simple_script)
+        pyc_opt1 = importlib.util.cache_from_source(
+            simple_script,
+            **compileall.optimization_kwarg(1)
+        )
+        pyc_opt2 = importlib.util.cache_from_source(
+            simple_script,
+            **compileall.optimization_kwarg(2)
+        )
+
+        compileall.compile_dir(path, quiet=True, optimize=[0, 1, 2], hardlink_dupes=True)
+
+        # import pdb; pdb.set_trace()
+
+        # All three files should have the same inode (hardlinks)
+        self.assertEqual(os.stat(pyc_opt0).st_ino, os.stat(pyc_opt1).st_ino)
+        self.assertEqual(os.stat(pyc_opt1).st_ino, os.stat(pyc_opt2).st_ino)
+
+        for pyc_file in set((pyc_opt0, pyc_opt1, pyc_opt2)):
```

Comment:
To improve readability, I suggest using the set literal, it works on all supported Python versions:

```suggestion
        for pyc_file in {pyc_opt0, pyc_opt1, pyc_opt2}:
```

Reply:
Done